node/src/daemon/mod.rs

```diff
@@ -795,6 +796,7 @@ mod tests {
 
     #[test]
     fn setup_judges_node_not_running_when_port_and_pid_are_set_but_os_says_different() {
+        running_test();
```

Comment:
`running_test()` and `clap_guard()` are not competitors. They do different things.

`running_test()` is for when you're going to do something that makes something call `exit_process()`. If nothing in your test will call `exit_process()`, then `running_test()` is meaningless. If it does, though, then `running_test` will make `exit_process()` call `panic!()`, which can be caught by the test, rather than `::std::process::exit()`, which cannot be caught by anything and brings the whole test run to a silent--and apparently successful--halt.

Calling `running_test()` is never harmful, though, as long as you don't call it from production code. It doesn't make anything inaccurate or slow.